src/gromacs/nblib/gmxsetup.cpp

```diff
@@ -215,56 +214,128 @@ std::unique_ptr<nonbonded_verlet_t> NbvSetupUtil::setupNbnxmInstance()
+    }
+    interactionConst.coulomb_modifier = eintmodPOTSHIFT;
+    interactionConst.rcoulomb         = options.pairlistCutoff;
+    // Note: values correspond to ic.coulomb_modifier = eintmodPOTSHIFT
+    interactionConst.dispersion_shift.cpot = -1.0 / gmx::power6(interactionConst.rvdw);
+    interactionConst.repulsion_shift.cpot  = -1.0 / gmx::power12(interactionConst.rvdw);
+
+    // These are the initialized values but we leave them here so that later
+    // these can become options.
+    interactionConst.epsilon_r  = 1.0;
+    interactionConst.epsilon_rf = 1.0;
+
+    /* Set the Coulomb energy conversion factor */
+    if (interactionConst.epsilon_r != 0)
+    {
+        interactionConst.epsfac = ONE_4PI_EPS0 / interactionConst.epsilon_r;
+    }
+    else
+    {
+        /* eps = 0 is infinite dieletric: no Coulomb interactions */
+        interactionConst.epsfac = 0;
+    }
 
-    gmxForceCalculator_p->nbv_ = setupNbnxmInstance();
+    calc_rffac(nullptr, interactionConst.epsilon_r, interactionConst.epsilon_rf,
+               interactionConst.rcoulomb, &interactionConst.k_rf, &interactionConst.c_rf);
+
+    if (EEL_PME_EWALD(interactionConst.eeltype))
+    {
+        // Ewald coefficients, we ignore the potential shift
+        interactionConst.ewaldcoeff_q = ewaldCoeff(1e-5, options.pairlistCutoff);
+        GMX_RELEASE_ASSERT(interactionConst.ewaldcoeff_q > 0, "Ewald coefficient should be > 0");
+        interactionConst.coulombEwaldTables = std::make_unique<EwaldCorrectionTables>();
+        init_interaction_const_tables(nullptr, &interactionConst);
+    }
 
-    // const PairlistSet& pairlistSet = nbv->pairlistSets().pairlistSet(gmx::InteractionLocality::Local);
-    // const gmx::index numPairs = pairlistSet.natpair_ljq_ + pairlistSet.natpair_lj_ + pairlistSet.natpair_q_;
-    // gmx_cycles_t cycles = gmx_cycles_read();
+    gmxForceCalculator_->interactionConst_ = std::move(interactionConst);
+}
 
-    matrix box_;
-    gmx::fillLegacyMatrix(system_.box().matrix(), box_);
+void NbvSetupUtil::setupForceRec(const matrix& box)
+{
+    gmxForceCalculator_->forcerec_.nbfp = nonbondedParameters_;
+    snew(gmxForceCalculator_->forcerec_.shift_vec, SHIFTS);
+    calc_shifts(box, gmxForceCalculator_->forcerec_.shift_vec);
+}
 
-    gmxForceCalculator_p->forcerec_.nbfp = nonbondedParameters_;
-    snew(gmxForceCalculator_p->forcerec_.shift_vec, SHIFTS);
-    calc_shifts(box_, gmxForceCalculator_p->forcerec_.shift_vec);
+void NbvSetupUtil::setParticlesOnGrid(const std::vector<gmx::RVec>& coordinates, const Box& box)
+{
+    gmxForceCalculator_->setParticlesOnGrid(particleInfoAllVdw_, coordinates, box);
+}
 
-    put_atoms_in_box(PbcType::Xyz, box_, system_.coordinates());
+void NbvSetupUtil::constructPairList(const gmx::ListOfLists<int>& exclusions)
+{
+    t_nrnb nrnb;
+    gmxForceCalculator_->nbv_->constructPairlist(gmx::InteractionLocality::Local, exclusions, 0, &nrnb);
```

Comment:
To support DD we will need to be able to pass InteractionLocality. Also, we don't have to use the gmx version, even in our translation layer, since we can just have the same enum order and get the correct int value.